core/loader/atom.ts

```diff
@@ -2,27 +2,37 @@ import type { TextResponse } from '../download.js'
 import type { OriginPost } from '../post.js'
 import { createPostsPage } from '../posts-page.js'
 import type { Loader } from './index.js'
-import { findAnchorHrefs, findLinksByType, toTime } from './utils.js'
+import {
+  findAnchorHrefs,
+  findLinksByType,
+  getImgReducer,
+  toTime
+} from './utils.js'
 
 function parsePosts(text: TextResponse): OriginPost[] {
   let document = text.parse()
   return [...document.querySelectorAll('entry')]
     .filter(entry => entry.querySelector('id')?.textContent)
-    .map(entry => ({
-      full: entry.querySelector('content')?.textContent ?? undefined,
-      intro: entry.querySelector('summary')?.textContent ?? undefined,
-      media: [],
-      originId: entry.querySelector('id')!.textContent!,
-      publishedAt: toTime(
-        entry.querySelector('published')?.textContent ??
-          entry.querySelector('updated')?.textContent
-      ),
-      title: entry.querySelector('title')?.textContent ?? undefined,
-      url:
-        entry
-          .querySelector('link[rel=alternate], link:not([rel])')
-          ?.getAttribute('href') ?? undefined
-    }))
+    .map(entry => {
+      const content = entry.querySelector('content')
+      const imageNodes = content?.querySelectorAll('img') ?? []
+      const mediaSet = [...imageNodes].reduce(getImgReducer('src'), new Set())
```

Comment:
Reducer logic is very complex and code is not really readable.

Maybe we can have `findImageBy(tags, attr)` method or something similar.